join.spec.md

```diff
@@ -100,204 +101,347 @@ flowchart TB
+
+## Root fields
+
+If this field appears at the root of the overall operation (query or mutation), then the field can just be placed at the root of the subgraph operation.
+
+```graphql example -- Root fields
+# Supergraph schema
+type Query {
+  fieldA: String @join__field(graph: A)
+  fieldB: String @join__field(graph: B)
+}
+
+# Operation
+{ fieldA fieldB }
+# Generated subgraph operations
+## On A:
+{ fieldA }
+## On B:
+{ fieldB }
 ```
 
-Documents MUST define a {join__Graph} enum. Each enum value defines a subgraph.
 
-:::[example](photos.graphql#join__Graph) -- Using join__Graph to define endpoints
+## Fields on the same subgraph as the parent operation
 
-The {join__Graph} enum is used as input to {@join} directives which link fields and types to subgraphs.
+If this field's parent field will be resolved by an operation on the same subgraph, then this field can be resolved as part of the same operation, by putting it in a nested selection set on the parent field's subgraph operation. Note that this example contains a {@join__owner} directive on an object type; this will be described later.
 
-# Directives
+```graphql example -- Fields on the same subgraph as the parent operation
+# Supergraph schema
+type Query {
+  fieldA: X @join__field(graph: A)
+}
 
-##! @join__graph
+type X @join__owner(graph: A) {
+  nestedFieldA: String @join__field(graph: A)
+}
 
-Declare subgraph metadata on {join__Graph} enum values. 
+# Operation
+{ fieldA { nestedFieldA } }
+# Generated subgraph operations
+## On A:
+{ fieldA { nestedFieldA }}
+```
 
-```graphql definition
-directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+## Fields provided by the parent field
+
+When resolving a field to a composite type (object, interface, or union), it is sometimes easy to calculate the values of nested fields inside that object even if those nested fields are ordinarily resolved in a different subgraph. When this is the case, you can include a `provides` argument in the `@join__field` listing these "pre-calculated" fields. The router can now resolve these fields in the "providing" subgraph instead of in the subgraph that would usually be used to resolve those fields.
+
+```graphql example -- Provided fields
+# Supergraph schema
+type Query {
+  fieldA: X @join__field(graph: A, provides: "usuallyBField")
+  fieldB: X @join__field(graph: B)
+}
+
+type X @join__owner(graph: B) {
+  usuallyBField: String @join__field(graph: B)
+}
+
+# Operation
+{ fieldB { usuallyBField } }
+# Generated subgraph operations
+## On B
+{ fieldB { usuallyBField } }
+
+# Operation
+{ fieldA { usuallyBField } }
+# Generated subgraph operations
+## On A
+{ fieldA { usuallyBField } }
```

Comment:
Differentiating these example operations with explanatory words could be helpful.

i.e.
typical resolution path
resolution path leveraging provides

Reply:
474f882